src/main/java/it/pagopa/transactions/mdcutilities/MDCContextLifter.java

```diff
@@ -58,14 +60,13 @@ private void copyToMdc(Context context) {
         if (!context.isEmpty()) {
             Map<String, String> mdcContextMap = Optional.ofNullable(MDC.getCopyOfContextMap()).orElseGet(HashMap::new);
             Map<String, String> reactorContextMap = context.stream()
+                    .filter(
+                            e -> Arrays.stream(TransactionTracingUtils.TracingEntry.values())
+                                    .anyMatch(tracingEntry -> tracingEntry.getKey().equals(e.getKey()))
+                    )
                     .collect(Collectors.toMap(e -> e.getKey().toString(), e -> e.getValue().toString()));
```

Comment:
```suggestion
             Map<String, String> reactorContextMap = Arrays
                    .stream(TransactionTracingUtils.TracingEntry.values())
                    .map(
                            key -> Tuples.of(
                                    key.toString(),
                                    context.getOrEmpty(key.toString()).map(Object::toString)
                                            .orElse(key.getDefaultValue())
                            )
                    )
                    .collect(
                            Collectors.toMap(
                                    TupleUtils.function(
                                            (
                                             k,
                                             v
                                            ) -> k
                                    ),
                                    TupleUtils.function(
                                            (
                                             k,
                                             v
                                            ) -> v
                                    )
                            )
                    );
```

iterate against `TransactionTracingUtils` enumeration instead of Reactor context to reduce for each operations to the only meaningful ones (avoid iterate against unneeded reactor context key here).

Reply:
or could use this version without tuples (uses SimpleEntry class)

```code
Map<String, String> reactorContextMap = Arrays
                    .stream(TransactionTracingUtils.TracingEntry.values())
                    .map(
                            key -> new AbstractMap.SimpleEntry<>(
                                    key.toString(),
                                    context.getOrEmpty(key.toString()).map(Object::toString)
                                            .orElse(key.getDefaultValue())
                            )
                    )
                    .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
```